template/web/app/paymaster-bundler/_components/PaymasterBundlerDemo.tsx

```diff
@@ -0,0 +1,131 @@
+import clsx from 'clsx';
+import { useAccount } from 'wagmi';
+import { useWriteContracts } from 'wagmi/experimental';
+import isLocal from '../../../src/utils/isLocal';
+import { usePaymasterBundlerContract } from '../_contracts/usePaymasterBundlerContract';
+import { CallStatus } from './CallStatus';
+
+// Use the local API URL to target the Paymaster directly without a proxy
+// if running on localhost, otherwise use the Paymaster Proxy.
+const paymasterURL = process.env.PAYMASTER_URL;
+const isLocalEnv = isLocal();
+const defaultUrl = isLocalEnv ? paymasterURL : `${document.location.origin}/api/paymaster-proxy`;
+
+export default function PaymasterBundlerDemo() {
+  const account = useAccount();
+  const { data: id, writeContracts } = useWriteContracts();
+
+  const contract = usePaymasterBundlerContract();
+
+  if (contract.status !== 'ready') {
+    console.error('Contract is not ready');
+    return null;
+  }
+
+  const handleMint = () => {
+    if (!account.address) {
+      alert('You need to be signed in to mint an NFT.');
```

Comment:
Why an `alert`? it feels so 2004

What are we trying to do here? Can we do it in a way it's more 2024.

Reply:
Updated to use react-toastify - much better and customizable

Reply:
<img width="1097" alt="Screenshot 2024-05-22 at 3 18 50 PM" src="https://github.com/coinbase/build-onchain-apps/assets/39774249/708698bc-d0a5-4bc7-a4e4-ba909cd2d453">